enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,933 @@
+and https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md for more detailed examples. -->
+
+TBD
+
+### API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Public subnet and outbound traffic
+
+It is a best practice to place the worker nodes in private subnets, although the
+AWS provides limited services in the Local Zones requiring a more complex setup to do so.
+
+Those are some options to create subnets on Local Zones:
+
+1) Create a subnet and associate it to a route table with the internet gateway as default.
+2) Create a private subnet, associating it to a route table from the zone in the
+  parent region. The cost will be twice as the traffic should cross the zone to
+  access the internet through a Nat Gateway, which also has fees to be applied.
+3) Create a private subnet, and provide access to the node through a proxy. Same costs
+  tradeoffs of option 2.
+
+Option 1 is covered by this enhancement to create the MachineSets (phase 1)
+and to automate by the installer (phase 2).
+
+The public subnet means the node will receive a Public IP address and the Security
+Group must not expose any ports to the internet. This must be the default behavior,
+and the security group for the `worker` pool will be used on the `edge` pool.
+
+> TODO: check if the current deployment is not exposing any port to the `0.0.0.0/0` on `worker` SG
+
+#### Infrastructure costs
+
+The internal data transfer costs (within the VPC: kubelet<->NLBs) will be the same
+of the other (normal workers) nodes[1][2].
+
+> *Data Transfer charges in AWS Local Zones is the same as in the Availability Zones today. Learn more about data transfer prices "in" and "out" of Amazon EC2 here.*
+
+The compute resource in Local Zones is more expensive, for example, `m5.xlarge` hosted
+in the parent zones of `us-west-2` will cost `$0.192/hr`, and in the LAX (Los Angeles)
+Local Zones is `$0.23/hr`, almost 17% more expensive than the regular zone.
+
+When following the architecture of using private subnets in Local Zones re-using Nat Gateway
+from the parent region (proposal #2 in the last section), considering NAT Gateway
+currently is not supported on the Local Zones locations, the following fees will be
+applied when the node needs access to the internet:
+
+- Fee for data transferred "in" to and "out" from NAT GW located in another AZ (`$0.01/GB` for each direction)
+- NAT GW fee per GB processed `$0.045`
+
+There is another cost of `$0.045/hr` for NAT GW allocation, which is not exclusive to a node in
+Local Zones as the Nat Gateway is already created to be used in a regular zone.
+
+When the node is in the public subnet (described in this EP), the fees applied will be the same to access the internet.
```

Comment:
This section is really great, would be good to include a real world example if that's something we have to hand, I'm thinking something like
"Given a workload that is processing x ingress data and y egress data, this would cost x on a normal zone and y on an edge zone", I don't know if we have some average data or whatever that we can suggest for a real world example though

Reply:
@JoelSpeed thanks.
I added one more paragraph to the "Infrastructure Costs" section trying to address and provide enough resources to address your feedback. I tried to be succinct providing resources to deepen the calculation of the scenarios as I was taking time to build many different hypothetical scenarios without enough data from real customers' use cases to enrich the content.

Reply:
https://github.com/openshift/enhancements/pull/1232/commits/7a2c770ce472992129114c223016379a80bd3f13